packages/sdk/src/utils/index.ts

```diff
@@ -0,0 +1,320 @@
+  BufferOrHex,
+  EncodeDecodeObjectOptions,
+  GetScriptTypeResponse,
+  IsBitcoinPaymentResponse,
+  NestedObject,
+  OneOfAllDataFormats,
+} from "./types";
+
+export function getNetwork(value: Network) {
+  if (value === "mainnet") {
+    return networks.bitcoin;
+  }
+
+  return networks[value];
+}
+
+export function createTransaction(
+  key: Buffer,
+  type: AddressType,
+  network: Network | BitcoinNetwork,
+  paymentOptions?: Payment,
+) {
+  initEccLib(ecc);
+  const networkObj =
+    typeof network === "string" ? getNetwork(network) : network;
+
+  if (type === "p2tr") {
+    return payments.p2tr({
+      internalPubkey: key,
+      network: networkObj,
+      ...paymentOptions,
+    });
+  }
+
+  if (type === "p2sh") {
+    return payments.p2sh({
+      redeem: payments.p2wpkh({ pubkey: key, network: networkObj }),
+      network: networkObj,
+    });
+  }
+
+  return payments[type]({ pubkey: key, network: networkObj });
+}
+
+export function getDerivationPath(
+  formatType: AddressFormat,
+  account = 0,
+  addressIndex = 0,
+) {
+  const pathFormat: Record<AddressFormat, string> = {
+    legacy: `m/44'/0'/${account}'/0/${addressIndex}`,
+    "p2sh-p2wpkh": `m/49'/0'/${account}'/0/${addressIndex}`,
+    segwit: `m/84'/0'/${account}'/0/${addressIndex}`,
+    taproot: `m/86'/0'/${account}'/0/${addressIndex}`,
+  };
+  return pathFormat[formatType];
+}
+
+export function hdNodeToChild(
+  node: BIP32Interface,
+  formatType: AddressFormat = "legacy",
+  addressIndex = 0,
+  account = 0,
+) {
+  const fullDerivationPath = getDerivationPath(
+    formatType,
+    account,
+    addressIndex,
+  );
+
+  return node.derivePath(fullDerivationPath);
+}
+
+export function toXOnly(pubkey: Buffer): Buffer {
+  return pubkey.subarray(1, 33);
+}
+
+export function tweakSigner(
+  signer: Signer,
+  opts: { tweakHash?: Buffer; network?: BitcoinNetwork } = {},
+): Signer {
+  const ECPair = ECPairFactory(ecc);
+
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  // @ts-ignore
+  let privateKey: Uint8Array | undefined = signer.privateKey!;
+  if (!privateKey) {
+    throw new OrditSDKError("Private key is required for tweaking signer!");
+  }
+  if (signer.publicKey[0] === 3) {
+    privateKey = ecc.privateNegate(privateKey);
+  }
+
+  const tweakedPrivateKey = ecc.privateAdd(
+    privateKey,
+    tapTweakHash(toXOnly(signer.publicKey), opts.tweakHash),
+  );
+  if (!tweakedPrivateKey) {
+    throw new OrditSDKError("Invalid tweaked private key!");
+  }
+
+  return ECPair.fromPrivateKey(Buffer.from(tweakedPrivateKey), {
+    network: opts.network,
+  });
+}
+
+export function tapTweakHash(pubKey: Buffer, h: Buffer | undefined): Buffer {
+  return crypto.taggedHash(
+    "TapTweak",
+    Buffer.concat(h ? [pubKey, h] : [pubKey]),
+  );
+}
+
+export const isObject = (o: unknown) => o?.constructor === Object;
+export const isString = (s: unknown) =>
+  s instanceof String || typeof s === "string";
+
+function encodeDecodeObject(
```

Comment:
@iamcrazycoder Is there an example object that is passed to this function?

Reply:
Nope. However, you can encode any string (make sure there special characters to see visual difference) and create an object out of it and use it.

```
{ foo: "<encoded-string>" }
```

Reply:
Added a test case for this and marked this function as deprecated since it's mutating the underlying object which will cause side effects. I will only modify the function after migrating all dependencies that use this.